packages/devtools/lib/src/memory/memory_inbounds.dart

```diff
@@ -0,0 +1,347 @@
+    final treeNode12 = InboundsTreeNode('class_1_2', 'field_c');
+    final treeNode13 = InboundsTreeNode('class_1_3', 'field_d');
+    final treeNode14 = InboundsTreeNode('class_4_4', 'field_e');
+    final treeNode15 = InboundsTreeNode('class_1_5', 'field_f');
+
+    final terryStuff = InboundsTreeNode(
+        'TerryStuff allocated TerryExtra', 'extra [object/14752]');
+    final shrineAppState1 = InboundsTreeNode('_ShrineAppState', '_stuff');
+    final shrineAppState2 = InboundsTreeNode('_ShrineAppState', '_stuff2');
+    final statefulElement1 = InboundsTreeNode('StatefulElement', 'state');
+    final hashmapEntry1 = InboundsTreeNode('_HashMapEntry', '_key');
+    final singleChildrenObjectElement1 =
+        InboundsTreeNode('SingleChildrenObjectElement', '_child');
+    final statefulElement2 = InboundsTreeNode('StatefulElement', 'state');
+    final hashmapEntry2 = InboundsTreeNode('_HashMapEntry', '_key');
+    final singleChildrenObjectElement2 =
+        InboundsTreeNode('SingleChildrenObjectElement', '_child');
+
+    data = InboundsTreeNode.root()
+      ..addChild(treeNode00
+        ..addChild(treeNode01)
+        ..addChild(treeNode02
+          ..addChild(treeNode03)
+          ..addChild(treeNode04)
+          ..addChild(treeNode05)))
+      ..addChild(treeNode10
+        ..addChild(treeNode11)
+        ..addChild(treeNode12)
+        ..addChild(treeNode13)
+        ..addChild(treeNode14)
+        ..addChild(treeNode15))
+      ..addChild(terryStuff
+        ..addChild(shrineAppState1
+          ..addChild(statefulElement1
+            ..addChild(hashmapEntry1..addChild(singleChildrenObjectElement1))))
+        ..addChild(shrineAppState2
+          ..addChild(statefulElement2
+            ..addChild(
+                hashmapEntry2..addChild(singleChildrenObjectElement2)))));
+  }
+
+  InboundsTreeNode data;
+}
+
+class InboundsTreeNode extends TreeNode<InboundsTreeNode> {
+  InboundsTreeNode(this._name, this._fieldName, [this.instanceHashCode]);
+  InboundsTreeNode.instance(this._instance, [this.instanceHashCode])
+      : _name = _instance.objectRef,
+        _fieldName = '';
+  InboundsTreeNode.root()
+      : _name = 'Instances',
+        _fieldName = '',
+        instanceHashCode = null;
+  InboundsTreeNode.empty()
+      : _name = null,
+        _fieldName = null,
+        instanceHashCode = null;
+
+  String _name;
+
+  InstanceSummary _instance;
+
+  InstanceSummary get instance => _instance;
+
+  String get name => _name;
+
+  /// isNew signals objectRef (object/###) has changed.
+  void setInstance(InstanceSummary theInstance, String hashCode,
+      [bool isNew = false]) {
+    _instance = theInstance;
+    instanceHashCode = hashCode;
+    _name = _name.split(' ')[0]; // Throw away instance objectRef name.
+    _name = (isNew && !isInboundEntry)
+        ? _instance.objectRef
+        : '$name (${instance.objectRef})';
+  }
+
+  final String _fieldName;
+
+  String get fieldName => _fieldName;
```

Comment:
The field is final - it can be public instead of wrapped in a getter.

https://dart.dev/guides/language/effective-dart/usage#prefer-using-a-final-field-to-make-a-read-only-property

Reply:
Good catch - done.